File: Lab_2_3_cpp/MyString.cpp
#include "MyString.h"
#include <string>
#include <cctype>

// Конструктор
MyString::MyString(std::string v) : value(v) {}

// Отримання рядка
std::string MyString::getValue() const {
    return value;
}

// Отримання довжини
int MyString::getLength() const {
    return (int)value.length();
}

// Переведення всього рядка у верхній регістр
void MyString::toUpper() {
    for (auto& c : value) {
        c = (char)toupper((unsigned char)c);
    }
}

File: Lab_2_3_cs/MyString.cs
namespace LabLibrary
{
    public class MyString
    {
        private string _value; // Приватне поле для зберігання тексту

        public string Value => _value; // Властивість для читання
        public int Length => _value.Length; // Властивість для отримання довжини

        public MyString(string value)
        {
            _value = value;
        }

        public void Upper()
        {
            _value = _value.ToUpper();
        }
    }
}